x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java

```diff
@@ -193,7 +192,7 @@ else if (p instanceof Lookup lookup) {
             checkBinaryComparison(p, failures);
             checkForSortableDataTypes(p, failures);
 
-            checkFilterMatchConditions(p, failures);
+            checkMatchQueryPredicates(p, failures);
```

Comment:
I understand that the `:` operator inherited some limitations similar to the other full text search functions, they don't have their compute engine yet. I feel like the more we expose similar interfaces, the more limitations we expose.  Perhaps having a compute engine is a final solution to these limitations. I wonder if there is any thought about pushing down these full text search related functions before some processing commands like `limit`, `stats` when possible? As the queries become complicated, these limitations are get more inconvenient.